### PR TITLE
Allow Pipeline to work with kwargs operations

### PR DESCRIPTION
**Pull request by ricardotealdi**
This PR will allow pipelined operations to work with inner operations that receives required properties as inputs (`property! :something`).

**EDIT:** It'll also rename `spec/active_operation/composed_operation_spec.rb` to `spec/active_operation/pipeline_spec.rb`

The following snippet works now:

```ruby
class SomePipeline < ActiveOperation::Pipeline
  class A < ActiveOperation::Base
    property! :language, accepts: [:de, :en], default: :en

    def execute
      case language
      when :de
        "Hallo Welt"
      when :en
        "Hello World"
      else
        raise "Unknown language"
      end
    end
  end

  class B < ActiveOperation::Base
    input :message

    def execute
      message
    end
  end

  use A
  use B
end

[2] pry(main)> SomePipeline.perform
=> "Hello World"
[3] pry(main)> SomePipeline.perform language: :de
=> "Hallo Welt"
```

### CODE REVIEW

File: lib/active_operation/pipeline.rb [hunk 1]
```diff
@@ -59,11 +63,15 @@ def inherited(subclass)
 
     protected
 
+    attr_reader :previous_output
```
Comment (t6d) [inline lib/active_operation/pipeline.rb]: Why is it necessary to store the previous output?

File: spec/active_operation/pipeline_spec.rb [hunk 2]
```diff
@@ -111,6 +125,26 @@ def execute
...
+  context(
+    "when composed of two operations, one that generates a string and one that lowercases strings by receiving its " \
+    "input as a required property"
+  ) do
+    subject(:composed_operation) do
+      operations = [string_generator, string_lowercaser_by_property]
+
+      Class.new(described_class) do
+        use operations.first
+        use operations.last, text: -> { previous_output }
```
Comment (t6d) [inline spec/active_operation/pipeline_spec.rb]: This contradicts the vision I have for pipelines. If you want Operation A to supply the keyword arguments for the subsequent Operation B, then I expect A to return a hash compatible with the properties of B.

Comment (t6d) [inline spec/active_operation/pipeline_spec.rb]: What I wanted is to support passing on pipeline arguments to an underlying operation:

```ruby
class Pipeline < ActiveOperation::Pipeline
  property! :transformation_method, accepts: [:lowercase, :uppercase], default: :lowercase

  use StringGenerator
  use StringTransformator, method: -> { transformation_method }
end
```

Comment (ricardotealdi) [inline spec/active_operation/pipeline_spec.rb]: Ohh sorry abou that. I misunderstood it! I'll try to change it!

Comment (ricardotealdi) [inline spec/active_operation/pipeline_spec.rb]: Just updated it

**Pull request merged**